tests/Functional/PipeTest.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * Copyright (C) 2019 by Jesus Franco Martinez <tezcatl@fedoraproject.org>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+namespace Functional\Tests;
+
+use function Functional\pipe;
+
+class PipeTest extends AbstractTestCase
+{
+    public function testPipeFunction()
```

Comment:
Could we use stock PHPUnit mock objects instead? I would prefer not to mix different mock implementations.

Reply:
It is not a great problem changing this to mocks, however I see that mock objects are deprecated and even its repo was archived already by Sebastian Bergmann, PHP 7.1 has reached EOL, too. What is your plan forward?

Reply:
Are you sure? My understanding was that the mock object package is deprecated but not the functionality as presented by PHPUnit (like `createMock()` etc.). I might be wrong though. I briefly checked Google but without success.

Reply:
The package `phpunit/phpunit-mock-objects` is deprecated/abandoned because the mocking functionality is directly integrated into `phpunit/phpunit`.

Reply:
See: 
- https://github.com/sebastianbergmann/phpunit/issues/3103
- https://github.com/sebastianbergmann/phpunit/blob/7.2.0/ChangeLog-7.2.md#changed

Reply:
Alright, fair. This still means using PHPUnit baked in mock objects is fine so let’s use it instead of prophecy

Reply:
finally found the time to drop the prophecy style using only mockObjects